### PR TITLE
Enable HOMEBREW_SORBET_RUNTIME (sometimes)

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Reverts #15385 with some tweaks:
- remove the commands that are now passing (on my machine)
- add some commands that are definitely failing (on my machine)
- provide the (temporary, undocumented) `HOMEBREW_NO_SORBET_RUNTIME` to disable Sorbet globally if needed more easily from e.g. specific GitHub Actions workflows 

# Please do not revert this PR! Instead: add the relevant commands to `NO_SORBET_RUNTIME_COMMANDS` or set `HOMEBREW_NO_SORBET_RUNTIME`.

Won't be merged until after 4.1.0 is out.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
@dduugg would love your help on some of these! Even before this PR is merged running `export HOMEBREW_SORBET_RUNTIME="1"` and then `brew generate-cask-api`, `brew generate-formula-api` or `brew release` and helping debug the errors would be great ❤️!

**Review by carlocab — APPROVED** [review]

**Comment by dduugg** [comment]
> @dduugg would love your help on some of these! Even before this PR is merged running `export HOMEBREW_SORBET_RUNTIME="1"` and then `brew generate-cask-api`, `brew generate-formula-api` or `brew release` and helping debug the errors would be great ❤️!

Happily! https://github.com/Homebrew/brew/pull/15710

File: Library/Homebrew/brew.sh
Comment (MikeMcQuaid) [inline Library/Homebrew/brew.sh:935]: @dduugg can any of these be removed now https://github.com/Homebrew/brew/pull/15710 is merged?
@Homebrew/maintainers @Bo98 if you have any other commands you know are failing with Sorbet: please shout so I can add in here before I merge (not today, post 4.1.0).

Comment (Bo98) [reply Library/Homebrew/brew.sh:935]: There was stuff in homebrew-core I think. I'll try dig into it tonight.

Comment (dduugg) [reply Library/Homebrew/brew.sh:935]: @MikeMcQuaid I believe these entries can be removed, I'm able to run these commands as of https://github.com/Homebrew/brew/pull/15710. (I tried the `generate*` commands with and without `-n`, and I tried `release` but stopped toying with it when I realized it was drafting releases in the `Homebrew` account 🙈).

Comment (MikeMcQuaid) [reply Library/Homebrew/brew.sh:935]:
```suggestion
```

**Review by carlocab — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks all!

**Pull request merged**
